src/android/InAppBrowser.java

```diff
@@ -1085,16 +1105,20 @@ public void onActivityResult(int requestCode, int resultCode, Intent intent) {
     public class InAppBrowserClient extends WebViewClient {
         EditText edittext;
         CordovaWebView webView;
+        boolean useBeforeload;
+        boolean waitForBeforeload;
 
         /**
          * Constructor.
          *
          * @param webView
          * @param mEditText
          */
-        public InAppBrowserClient(CordovaWebView webView, EditText mEditText) {
+        public InAppBrowserClient(CordovaWebView webView, EditText mEditText, boolean useBeforeload) {
             this.webView = webView;
             this.edittext = mEditText;
+            this.useBeforeload = useBeforeload;
```

Comment:
Why 2 vars that will always store the same value?

Reply:
(this is merged already, but perhaps my reply will serve as a reference)
`waitForBeforeload` is the current state, whether we need to wait a call back from JS to really load next time it is encountered. `useBeforeload` indicates whether the `beforeload` event is used at all (see the end of `shouldOverrideUrlLoading`).